lib/signalfx/tracing/sfx_logger.rb

```diff
@@ -0,0 +1,58 @@
+require 'logger'
+require 'fileutils'
+
+module SignalFx
+  module Tracing
+    module Logging
+
+      LOG_LEVELS = {
+          'unknown' => Logger::UNKNOWN,
+          'fatal' => Logger::FATAL,
+          'error' => Logger::ERROR,
+          'warn' => Logger::WARN,
+          'info' => Logger::INFO,
+          'debug' => Logger::DEBUG
+      }.freeze
+
+      DEFAULT_LOG_PATH = '/var/log/signalfx/signalfx-ruby-tracing.log'
+      DEFAULT_SHIFT_AGE = 5
+      DEFAULT_SHIFT_SIZE = 1048576
+
+      def self.create (log_path = ENV['SIGNALFX_LOG_PATH'] || DEFAULT_LOG_PATH,
+                       sfx_shift_age = ENV['SIGNALFX_LOG_SHIFT_AGE'] || DEFAULT_SHIFT_AGE,
+                       sfx_shift_size = ENV['SIGNALFX_LOG_SHIFT_SIZE'] || DEFAULT_SHIFT_SIZE)
+
+        log_level = ENV['SIGNALFX_LOG_LEVEL'].downcase if ENV['SIGNALFX_LOG_LEVEL']
+        if log_path == 'STDERR'  || log_path == 'STDOUT'
+          self.create_std_logger('log_path')
+        else
+            begin
+              @logger = Logger.new("#{log_path}", shift_age = sfx_shift_age, shift_size = sfx_shift_size)
+            rescue Errno::ENOTSUP => e
+              self.create_std_logger('STDERR', log_path, e)
+            rescue Errno::EACCES => e
+              self.create_stderr_logger('STDERR', log_path, e)
```

Comment:
is this a function?

Reply:
I was still in the process of cleaning up the code. Cleaned it up.